analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/EnvironmentalVariableScope.cs

```diff
@@ -0,0 +1,84 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+
+namespace SonarAnalyzer.UnitTest.Helpers
+{
+    /// <summary>
+    /// Defines a scope inside which new environment variables can be set.
+    /// The variables will be cleared when the scope is disposed.
+    /// Copy-pasted from S4NET codebase.
+    /// </summary>
+    public sealed class EnvironmentVariableScope : IDisposable
+    {
+        private IDictionary<string, string> originalValues = new Dictionary<string, string>();
+
+        public void SetVariable(string name, string value)
+        {
+            // Store the original value, or null if there isn't one
+            if (!originalValues.ContainsKey(name))
+            {
+                originalValues.Add(name, Environment.GetEnvironmentVariable(name));
+            }
+            Environment.SetEnvironmentVariable(name, value, EnvironmentVariableTarget.Process);
+        }
+
+        public void SetVariableInRelease(string name, string value)
+        {
+            // do nothing in Debug mode
+#if RELEASE
```

Comment:
This should use `TestContextHelper.IsAzureDevOpsContext`, same as `ParseOptionsHelper`

Reply:
How is it connected with Azure at all? This is just for local debugging purposes.

Reply:
Same as the IsAzureDevOpsContext is there to behave differently under CI and local debugging purposes.
On CI, we run all language versions.
On local machine (maybe except release test run from a script) we run only one language version to make it faster and easier (to avoid hitting the same breakpoint 4 times, actually)

The key point is to have only one common mechanism to deal with the same task `I want easy debugging on localhost`.

Reply:
But that won't help us detecting issues with concurrency on local runs, unless you remember and set this property explicitly for that every time you modify the rule.